src/main/java/com/conveyal/taui/controllers/RegionalAnalysisController.java

```diff
@@ -79,34 +83,107 @@ public static RegionalAnalysis deleteRegionalAnalysis (Request req, Response res
     }
 
     /**
-     * Get a particular percentile of a query as a grid file.
-     * FIXME this does not seem to do that anymore. It just gets the single percentile that exists for any one analysis.
+     * The client usually fetches gzipped grids of regional analysis results from S3 when the analysis is done.
+     * This endpoint allows fetching the partially completed grid from the backend while the analysis is still in
+     * progress.
      */
-    public static Object getPercentile (Request req, Response res) throws IOException {
+    public static InputStream getPartialRegionalAnalysisResult (Request req, Response res) {
+        RegionalAnalysis analysis = Persistence.regionalAnalyses.findByIdFromRequestIfPermitted(req);
+        if (analysis == null) {
+            res.status(404);
+            res.body("The specified regional analysis does not exist.");
+            return null;
+        }
+        if (analysis.complete || analysis.deleted) {
```

Comment:
Deleted should be a 404.

Reply:
Also, if it's complete then wouldn't a 2xx/3xx response be more appropriate?

Reply:
This is in dead code, but I'm moving it over to the endpoint we actually use. Analysis.complete doesn't tell us anything (see inline comment in this PR) so we're only dealing with deleted. Strictly speaking I think 410 is the correct code here ("410 Gone: The requested resource is no longer available at the server and no forwarding address is known. This condition is expected to be considered permanent.") But do you prefer a 404?

We should definitely be returning 20x for any response on a completed job.

Reply:
Actually I'll just use AnalysisServerException.notFound() to reply 404 for both situations (deleted / never existed), it's simpler.

Reply:
Fixed in bf2a54b